domain/removal/state/model/model.go

```diff
@@ -245,6 +246,46 @@ func (st *State) GetModelLife(ctx context.Context, mUUID string) (life.Life, err
 	return life, errors.Capture(err)
 }
 
+// MarkModelAsDead marks the model with the input UUID as dead.
+// If there are model dependents, then this will return an error.
+func (st *State) MarkModelAsDead(ctx context.Context, mUUID string) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	modelUUID := entityUUID{UUID: mUUID}
+	updateStmt, err := st.Prepare(`
+UPDATE model_life
+SET    life_id = 2
+WHERE  model_uuid = $entityUUID.uuid
+AND    life_id = 1`, modelUUID)
+	if err != nil {
+		return errors.Errorf("preparing model life update: %w", err)
+	}
+	return errors.Capture(db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if l, err := st.getModelLife(ctx, tx, mUUID); err != nil {
```

Comment:
Same as before. Give it a shot first.

Reply:
I prefer the valid error message. Considering how often this is run, I think it's just easier and more logical to do it this way.